packages/filecoin-wallet-provider/src/filecoin.ts

```diff
@@ -312,18 +303,19 @@ export class Filecoin {
   getReplaceMessageMinGasParams = async (
     message: LotusMessage,
   ): Promise<{ gasFeeCap: string; gasPremium: string; gasLimit: number }> => {
-    let newFeeCap = message.GasFeeCap
-    const newPremium = new BigNumber(message.GasPremium)
-      .multipliedBy(125)
-      .dividedBy(100)
 
-    if (newPremium.isGreaterThan(message.GasFeeCap)) {
-      newFeeCap = newPremium.toFixed(0, BigNumber.ROUND_CEIL)
-    }
+    const newPremiumBn = new BigNumber(message.GasPremium)
+      .times(1.25)
+      .plus(Number.EPSILON)
```

Comment:
So `Number.EPSILON` acts somewhat dynamically in this case?

Reply:
It only ensures that whole values that result from `message.GasPremium * 1.25` are rounded to the next whole integer.

So `100 * 1.25` becomes `125`, rounded to `126`
While `99 * 1.25` becomes `123,75`, rounded to `124`

Another stragegy might be to always add `1` before rounding
So `100 * 1.25 + 1` becomes `125`, rounded to `126`
And `99 * 1.25 + 1` becomes `124,75`, rounded to `125`

So the difference is not too great, we only ensure we round everything to the next whole integer, even if the result was a whole integer to begin with